TrafficCapture/captureOffloader/src/main/java/org/opensearch/migrations/trafficcapture/StreamChannelConnectionCaptureSerializer.java

```diff
@@ -276,8 +280,11 @@ private void addStringMessage(int captureFieldNumber, int dataFieldNumber,
         writeByteStringToCurrentStream(dataFieldNumber, str);
     }
 
-    private void addDataMessage(int captureFieldNumber, int dataFieldNumber, Instant timestamp, ByteBuf buffer) throws IOException {
-        var byteBuffer = buffer.nioBuffer();
+    void addDataMessage(int captureFieldNumber, int dataFieldNumber, Instant timestamp, ByteBuf buf) throws IOException {
```

Comment:
Moved these to package-private to allow for unit testing with nio ByteBuffers to verify some underlying behavior that was easier to do than through Netty ByteBufs

Reply:
What i wanted to test specifically was behavior if ByteBuf.nioBuffer returns a buffer with Capacity != Limit.

I had difficulty testing this, because it only happens in very specific cases.. i had only theorized it until now where i've been able to produce the case:

```
        var nettyBuf = Unpooled.wrappedBuffer(byteBuffer.asReadOnlyBuffer());
        nettyBuf.writerIndex(nettyBuf.writerIndex() - 1);
        var nioBuffer = nettyBuf.nioBuffer();
        Assertions.assertTrue(nioBuffer.limit() < nioBuffer.capacity());
```

Would you prefer this?

Reply:
Fixed this with a further refactor that cleaned up and consolidated the interface between Netty ByteBufs and nio ByteBuffers